src/api/localResolvers/experiment.js

```diff
@@ -137,64 +155,71 @@ export default () => {
 			Query: {
 				experiment(_, { id }, context) {
 					// get the existing assigned version for this experiment id
-					let version = assignments[id];
+					let currentAssignment = assignments[id] || {};
 
 					// read the experiment data from the cache
 					const experiment = readJSONSetting(context, `cache.data.data['Setting:uiexp.${id}'].value`);
+					// get the hash for our current experiment setting
+					const settingHash = hashCode(JSON.stringify(experiment));
```

Comment:
Should we maybe only hash the weights? I'm thinking that we don't necessarily want to force a reassignment just because dates change or the experiment gets disabled

Reply:
hmmm... That's prob a good call...I'll work on excluding the date change and enabled field. The cookie cleaner with remove disabled experiments as they won't be in the active list anymore.

I'll think about how to add a way to trigger a general reassignment of the experiment to cover allowing those fields to remain. Maybe just a simple incrementing value field in the settings object...